core/service/src/lib.rs

```diff
@@ -717,6 +719,20 @@ fn build_network_future<
 						let _ = sender.send(network_state);
 					}
 				}
+				rpc::system::Request::NodeRole(sender) => {
+					use rpc::system::NodeRole;
+
+					let role = match roles {
```

Comment:
That really won't work all that well. Usually the `roles` will be a combination of those. Why not return a vector in the RPC as well?

Reply:
Can a light client be an authority? If not, I think it's best that we just return a single role, because authority should imply that it's also a full node. Maybe I'm just being confused by this code though (which might not be fully accurate):

https://github.com/paritytech/substrate/blob/00ac65fca22652807e578326ad9f2fe6b169390a/core/network/src/config.rs#L102-L117

I'll try and change the code to better check bitmasks though.